partiql-eval/src/main/java/org/partiql/eval/PartiQLEvaluator.java

```diff
@@ -0,0 +1,68 @@
+package org.partiql.eval;
+
+import org.jetbrains.annotations.NotNull;
+import org.partiql.plan.Plan;
+import org.partiql.spi.value.Datum;
+
+/**
+ * TODO JAVADOC
+ */
+public interface PartiQLEvaluator {
```

Comment:
I'm not convinced we need this -- since the `StandardEvaluator` implementation is a single line.

This is a "core" package. I'm of the opinion that some other package _could_ provide easier-to-use APIs. If this is the intention, maybe in `partiql-lang`, we expose an API that takes in a `String query` and returns the result.

Reply:
The standard may be a single line, but consider introducing a plan cache. It's not necessarily about an easier API, but about splitting the responsibility of the preparation phase from invoking.

The compiler returns a function, the evaluator invokes that function. I could put these both into PartiQLEvaluator, but then we'd have to extend the "evaluator" with compilation strategies for choosing physical operators.

So it's really not about a "easier API" but a responsibility split, where the default custom need not interact with the compilation phase.

Reply:
> It's not necessarily about an easier API, but about splitting the responsibility of the preparation phase from invoking... The compiler returns a function, the evaluator invokes that function.

In this implementation, the PartiQLEvaluator doesn't invoke the function returned by the PartiQLCompiler. The responsibilities here overlap. The PartiQLEvaluator here compiles the plan and executes. The PartiQLCompiler here compiles the plan and does not execute.

Reply:
The compiler returns a compiled "function" aka a prepared statement to be invoked via "execute". The PartiQLEvaluator does call the "function/statement" returned by the compiler. 

The responsibility of the evaluator indeed overlaps the compiler by design. The split exists so that customers can supply their own compilation phase, but don't need to. 

The idea here was to make an entry-point for custom compilation, but independent of the evaluator. Do you have some ideas on how to best split these responsibilities?

We could also make evaluator drive the datum for customers who don't want the lazy behavior?